eclipse-collections/src/main/java/org/eclipse/collections/impl/map/mutable/MapAdapter.java

```diff
@@ -163,6 +164,31 @@ public V removeKey(K key)
         return this.delegate.remove(key);
     }
 
+    @Override
+    public boolean removeAllKeys(Set<? extends K> keys)
+    {
+        Objects.requireNonNull(keys);
+        int previousSize = this.size();
+
+        if (previousSize > keys.size())
+        {
+            keys.forEach(this::removeKey);
+        }
+        else
+        {
+            Iterator<K> iterator = this.keySet().iterator();
+            while (iterator.hasNext())
+            {
+                K key = iterator.next();
+                if (keys.contains(key))
+                {
+                    iterator.remove();
+                }
+            }
+        }
+        return previousSize > this.size();
```

Comment:
This one can be removed

Reply:
why can it be or rather should it be removed? Because of the default method?

Reply:
Yes because of the default method. Also an adapter could be adapting a ConcurrentHashMap. In an adapter, we can't assume the optimization is safe to do since we can't assume anything about the nature of the adapted map.

Reply:
Ok makes sense

Reply:
Ok makes sense